autorag/evaluation/metric/retrieval.py

```diff
@@ -1,67 +1,52 @@
-import functools
 import itertools
-from typing import List
-
 import math
 
-from autorag.utils.util import convert_inputs_to_list
-
-
-def retrieval_metric(func):
-	@functools.wraps(func)
-	@convert_inputs_to_list
-	def wrapper(
-		retrieval_gt: List[List[List[str]]], pred_ids: List[List[str]]
-	) -> List[float]:
-		results = []
-		for gt, pred in zip(retrieval_gt, pred_ids):
-			if gt == [[]] or any(bool(g_) is False for g in gt for g_ in g):
-				results.append(None)
-			else:
-				results.append(func(gt, pred))
-		return results
+from autorag.evaluation.metric.util import autorag_metric
+from autorag.schema.metricinput import MetricInput
 
-	return wrapper
 
-
-@retrieval_metric
-def retrieval_f1(gt: List[List[str]], pred: List[str]):
+@autorag_metric(fields_to_check=["retrieval_gt", "retrieval_ids"])
+def retrieval_f1(metric_input: MetricInput):
 	"""
 	Compute f1 score for retrieval.
 
-	:param gt: 2-d list of ground truth ids.
-	    It contains and/or connections between ids.
-	:param pred: Prediction ids.
+	:param metric_input: The MetricInput schema for AutoRAG metric.
 	:return: The f1 score.
 	"""
-	recall_score = retrieval_recall.__wrapped__(gt, pred)
-	precision_score = retrieval_precision.__wrapped__(gt, pred)
+	recall_score = retrieval_recall.__wrapped__(metric_input)
+	precision_score = retrieval_precision.__wrapped__(metric_input)
 	if recall_score + precision_score == 0:
 		return 0
 	else:
 		return 2 * (recall_score * precision_score) / (recall_score + precision_score)
 
 
-@retrieval_metric
-def retrieval_recall(gt: List[List[str]], pred: List[str]):
+@autorag_metric(fields_to_check=["retrieval_gt", "retrieval_ids"])
+def retrieval_recall(metric_input: MetricInput) -> float:
+	gt, pred = metric_input.retrieval_gt, metric_input.retrieval_ids
```

Comment:
I think you need to check `metric_input.retrieval_gt`
There was feature like this before

```python
def wrapper(
		retrieval_gt: List[List[List[str]]], pred_ids: List[List[str]]
	) -> List[float]:
		results = []
		for gt, pred in zip(retrieval_gt, pred_ids):
			if gt == [[]] or any(bool(g_) is False for g in gt for g_ in g):
				results.append(None)
```

But you deleted it.

Reply:
is_fields_notnone method can replace that functionality. in my think